elements-fun/src/wally.rs

```diff
@@ -100,11 +101,11 @@ pub fn bip39_mnemonic_to_seed(mnemonic: &str, passphrase: &str) -> Option<[u8; B
 
 /// Calculate the signature hash for a specific index of
 /// an Elements transaction.
-pub fn tx_get_elements_signature_hash(
+pub fn tx_get_elements_signature_hash<V: Encodable>(
     tx: &crate::Transaction,
     index: usize,
     script_code: &bitcoin::Script,
-    value: &crate::confidential::Value,
+    value: &V, // TODO: Is this really the best way to do it?
```

Comment:
:question: Should we introduce a `ValueCommitment` type?

Reply:
It actually exists. Should we use it here?

Reply:
Well, you can use this function for confidential and non-confidential values. The new type system doensn't offer anything here.